### PR TITLE
Add Cucumber-JVM migration module

### PR DESCRIPTION
**Pull request by timtebeek**
Initial implementation based on export of [rewrite-testing-framwork cucumber recipes](https://github.com/openrewrite/rewrite-testing-frameworks/tree/main/src/main/java/org/openrewrite/java/testing/cucumber) with permission as both the original author there, and an employee of @moderneinc . 

This project aims to help users of cucumber-jvm migrate to newer versions by changing their code to adopt breaking changes.

### CODE REVIEW

File: .github/workflows/release-github.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    name: Create GitHub Release and Git tag
+    runs-on: ubuntu-latest
+    environment: Release
+    permissions:
+      contents: write
+    steps:
+      - uses: actions/checkout@v3
+      - uses: cucumber/action-create-github-release@v1.1.0
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (mpkorstanje) [inline .github/workflows/release-github.yml:18]: Note to self: Add this to the secrets.

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+        <developerConnection>scm:git:git@github.com:cucumber/cucumber-jvm-migration.git</developerConnection>
+        <url>git://github.com/cucumber/cucumber-jvm-migration.git</url>
+        <tag>HEAD</tag>
+    </scm>
+
+    <dependencies>
+        <dependency>
+            <groupId>io.cucumber</groupId>
+            <artifactId>cucumber-java</artifactId>
+            <version>${cucumber.version}</version>
```
Comment (mpkorstanje) [inline pom.xml]: This could be replaced with the cucumber bom.

File: pom.xml [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+        </dependency>
+        <dependency>
+            <groupId>io.cucumber</groupId>
+            <artifactId>cucumber-junit-platform-engine</artifactId>
+            <version>${cucumber.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.junit.platform</groupId>
+            <artifactId>junit-platform-suite-api</artifactId>
+            <version>1.9.2</version>
```
Comment (mpkorstanje) [inline pom.xml]: This can be replaced with the junit bom.

File: pom.xml [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+        </dependency>
+
+        <dependency>
+            <groupId>org.openrewrite</groupId>
+            <artifactId>rewrite-java</artifactId>
+            <version>${rewrite.version}</version>
```
Comment (mpkorstanje) [inline pom.xml]: This can use the rewrite-bom

File: pom.xml [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <configuration>
+                    <source>${base.java.version}</source>
+                    <target>${base.java.version}</target>
+                    <!-- Allow for text blocks in tests -->
+                    <testSource>17</testSource>
+                    <testTarget>17</testTarget>
```
Comment (mpkorstanje) [inline pom.xml:173]: That's new. Cool.

File: pom.xml [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+                    <value>true</value>
+                </property>
+            </activation>
+            <build>
+                <pluginManagement>
+                    <plugins>
+                        <!-- Check code format -->
+                        <plugin>
+                            <groupId>com.diffplug.spotless</groupId>
+                            <artifactId>spotless-maven-plugin</artifactId>
```
Comment (mpkorstanje) [inline pom.xml]: Let's drop this . This project isn't so that it would benefit from the hassle.

File: pom.xml [hunk 7]
```diff
@@ -0,0 +1,240 @@
...
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-resources-plugin</artifactId>
+                <configuration>
+                    <encoding>UTF-8</encoding>
```
Comment (mpkorstanje) [inline pom.xml]: Should be a property.

File: .github/stale.yml [hunk 8]
```diff
@@ -0,0 +1,17 @@
+# Number of days of inactivity before an issue becomes stale
+daysUntilStale: 300
```
Comment (mpkorstanje) [inline .github/stale.yml]: Can be removed. Stale bot was discontinued.

File: .github/lock.yml [hunk 9]
```diff
@@ -0,0 +1,34 @@
+# Configuration for lock-threads - https://github.com/dessant/lock-threads
+
+# Number of days of inactivity before a closed issue or pull request is locked
+daysUntilLock: 365
```
Comment (mpkorstanje) [inline .github/lock.yml]: Can be removed. Stale bot was discontinued.

**Review by mpkorstanje — COMMENTED** [review]
I'll try and pass for fixing the problems. Mostly small fry issues.

**Comment by mpkorstanje** [comment]
@timtebeek I won't get around to attending these changes today. If you happen to have time in the 14 days do feel free.

File: pom.xml (hunk 5 above)
Comment (timtebeek) [reply pom.xml:173]: Yes was new to me as well; very handy when you want to declare before/after state in recipes as Strings.

**Comment by timtebeek** [comment]
Thanks for reviewing; applied your suggestions just now.

Had the version set for `0.0.1-SNAPSHOT`, but maybe `0.1.0-SNAPSHOT` would be better; what do you think?

**Comment by mpkorstanje** [comment]
Yup `0.1.0-SNAPSHOT`.  First feature release of an experimental module.

**Review by mpkorstanje — APPROVED** [review]
Looks good! I'll still have to setup the release environment to make the first release. Cheers!

**Comment by timtebeek** [comment]
Thanks for the merge! Once it's released I'll pick it up on our end to make it available in the platform, and see if there's any projects I can use to try it out.

**Pull request merged**